### PR TITLE
Issues/961/rbac tcerts no attrs, 

### LINKED ISSUE #1851: Peer2 doesn't restart communication with Peer1 once the communication breaks
## Description

When communication breaks between Peer1 & Peer2 due to a network glitch, Peer2 doesn't reconnect to Peer1 and stays idle.
## Describe How to Reproduce

1. Start a network with 2 peers
2. Break the communication between the two peers
3. Peer 2 doesn't reconnect to Peer 1
## Proposal

Ref: https://github.com/masterDev1985/fabric/commit/6adb13c053b14ab19ace0d095806046bc2086724
In the communication loop in chatPeer() function, continue on an error - thus reconnecting.

Note: In the above solution, the sleep between reconnects should be configurable from yaml file & there should be a maximum number of retries (or handle the "duplicate handler" error differently).

### PR DESCRIPTION
**Pull request by jeffgarratt**
## Description

Adds ability to test RBAC based chaincode through enhancements to behave user registration mechanism.  Also added Python generated GRPC stubs for all services and utilize these stubs for the chaincode to chaincode behave test.  Added coverage report capability for behave testing, with aggregate calculation.  Can be tuned based upon desired coverage.  Added simple touch function to Peer to ensure reconnect on total network disconnect.   
## Motivation and Context

These changes are required to enhance the capability of behave testing wrt to RBAC and to lay the groundwork for moving from HTTP based testing to GRPC.

Fixes #1851 #961 
## How Has This Been Tested?

New behave tests have been added and tested locally, but they are disabled pending an image update for the grpcio python library.
## Checklist:
- [X] I have added a [Signed-off-by](https://github.com/hyperledger/fabric/blob/master/CONTRIBUTING.md#legal-stuff)
- [X] Either no new documentation is required by this change, OR I added new documentation
- [X] Either no new tests are required by this change, OR I added new tests
- [X] I have run [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports), [go vet](https://golang.org/cmd/vet/), and [golint](https://github.com/golang/lint). I have cleaned up all valid errors and warnings in code I have added or modified. These tools may generate false positives. Don't be worried about ignoring some errors or warnings. The goal is clean, consistent, and readable code.

Signed-off-by: Jeff Garratt garratt.jeff@gmail.com

### CODE REVIEW

**Comment by jeffgarratt** [comment]
@binhn here is the PR binhn.  I would ask that you run this locally first, as I want to make sure that you can pass the behave and unit tests without issue.  I have disabled the new tests that rely on the new grpc stubs until I can coordinate with @ghaskins to make sure the image creation is updated.

**Comment by binhn** [comment]
@jeffgarratt Linter errors:

```
examples/chaincode/go/rbac_tcerts_no_attrs/rbac.go:49: missing verb at end of format string in Errorf call
examples/chaincode/go/rbac_tcerts_no_attrs/rbac_test.go:158: possible formatting directive in Fatal call
examples/chaincode/go/rbac_tcerts_no_attrs/rbac_test.go:430: missing verb at end of format string in Errorf call
exit status 1
make: *** [linter] Error 1
```

**Comment by binhn** [comment]
@jeffgarratt I ran `make unit-test` locally and failed:

```
ok      github.com/hyperledger/fabric/tools/dbutility   0.057s  coverage: 32.4% of statements
Stopping peer..
make: *** [unit-test] Error 1
```

due to:

```
2016/06/18 13:17:36 Is db path [/var/hyperledger/test/tmpdb/db] empty [false]
Error opening DB IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable
--- FAIL: TestExecuteDeploySysChaincode (0.00s)
panic: Could not open openchain db error = [IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable] [recovered]
    panic: Could not open openchain db error = [IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable]
```

which @jyellick fixed in 1877 that I've just merged. So I think if you rebase, it should resolve this error.

File: core/peer/peer.go [hunk 1]
```diff
@@ -478,6 +478,12 @@ func (p *Peer) GetPolicyManager(cid string) policies.Manager {
 	return nil
 }
 
+// GetCreateFromSnapshotInfo queries ledger mgr to get information for any in-progress ledger creation
+// from a snapshot.
+func (p *Peer) GetCreateFromSnapshotInfo() string {
+	return p.LedgerMgr.GetCreateFromSnapshotInfo()
+}
+
```
Comment (manish-sethi) [inline core/peer/peer.go]: Can you rename this function to `JoinBySnaphotStatus` - at both the places (in the peer and in the LedgerMgr). Also, I tend to prefer to return a struct from `LedgerMgr` that makes it self-explaining. The struct (`JoinBySnapshot`) can have two fields - `{InProgress bool, BootSnapshotDir string}`.

**Pull request closed without merging**